### PR TITLE
APS-856 - Add placement request match outcome report v2

### PR DESCRIPTION
**Pull request by davidatkinsuk**

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/cas1/Cas1PlacementMatchingOutcomesV2ReportTest.kt [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+
+    standardRFPNotAssessed.createRequestForPlacement()
+  }
+
+  @Test
+  fun `Get report is empty if no applications`() {
+    `Given a User`(roles = listOf(UserRole.CAS1_REPORT_VIEWER)) { _, jwt ->
+
+      webTestClient.get()
+        .uri(getReportUrl(year = 2019, month = 2, includePii = true))
```
Comment (edavey) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/cas1/Cas1PlacementMatchingOutcomesV2ReportTest.kt:97]: would the setup be clearer here if you also used constants, as you do with the "correct applications" example?

e.g. `.uri(getReportUrl(year = BLANK_YEAR, month = BLANK_MONTH, includePii = true))`

**Review by edavey — APPROVED** [review]
Looks good to me. I haven't tested the SQL myself, but am very interested to see how you're tackling the complexity, e.g. with the little `MonthSpecificReportParams` helper class and the clear use of common table expressions. 🏅

File: src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/cas1/Cas1PlacementMatchingOutcomesV2ReportTest.kt (hunk 1 above)
Comment (davidatkinsuk) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/cas1/Cas1PlacementMatchingOutcomesV2ReportTest.kt:97]: Good idea, i'll change that

FYI - i should have left this PR as draft as SQL is still work in progress (apologies), but TBH i think you've reviewed the key changes as the additional SQL will just result in more test code

**Comment by davidatkinsuk** [comment]
Hi @edavey i appreciate you approved this before but that point the SQL wasn't complete. This PR is now completed and ready to merge. Are you happy for me to merge? All i've done since last time is added a bunch of SQL and more integration tests.

**Comment by edavey** [comment]
> Hi @edavey i appreciate you approved this before but at that point the SQL wasn't complete. This PR is now completed and ready to merge. Are you happy for me to merge? All i've done since last time is added a bunch of SQL and more integration tests.

Yes this looks good to me, merge away, thks!

**Pull request merged**